ghc92/Prelude/Linear/Internal/TypeEq.hs

```diff
@@ -0,0 +1 @@
+module Prelude.Linear.Internal.TypeEq where
```

Comment:
Add documentation for why this module exists, and explaining that when the module is removed, then we also should remove the corresponding ` {-# OPTIONS_GHC -Wno-dodgy-exports -Wno-unused-imports #-}`.